.github/actions/runTestsTaggedAs/action.yaml

```diff
@@ -18,10 +18,15 @@ runs:
         distribution: 'zulu'
         java-version: 11
     - name: Cache Maven repository
-      uses: actions/cache@v3
+      id: cacheMavenArtifacts
+      uses: actions/cache/restore@v3 #use restore action in order to skip post-action that saves cache
       with:
         path: ~/.m2/repository
         key: ${{ github.run_id }}-${{ github.run_number }}-maven-cache
+    - name: Maven artifacts creation #if for some reason there was a cache miss then create the maven artifacts
```

Comment:
This would fall back to the case of each test job to run the entire build independently - are we sure that's preferrable to simply failing the job?
As you are protecting from cache-get failures, reattempting the individual job again would probably be faster?

Reply:
I'd suggest introducing the cache retrieval timeout explained in https://github.com/actions/cache/blob/main/tips-and-workarounds.md#cache-segment-restore-timeout - or was that already in place?

Reply:
the timeout was already in place (10 minutes). I honestly don't know in that case when the platform will be available in the future so I preferred to insert the building of the artifacts instead of risking losing other 10 minutes in a retry of the cache retrieval (if you think that building the artifacts last something like 10 minutes)